lib/manageiq/api/open_api/generator.rb

```diff
@@ -0,0 +1,113 @@
+module ManageIQ
+  module Api
+    module OpenApi
+      class Generator
+        require 'json'
+
+        OPENAPI_VERSION = "3.0.0".freeze
+        PARAMETERS_PATH = "/components/parameters".freeze
+        SCHEMAS_PATH    = "/components/schemas".freeze
+
+        attr_reader :manageiq_api_path, :openapi_path, :openapi_spec
+
+        def initialize
+          manageiq_api_engine = Vmdb::Plugins.all.detect { |e| e.name == "ManageIQ::Api::Engine" }
+
+          @manageiq_api_path = manageiq_api_engine.root
+          @openapi_path      = manageiq_api_path.join("config", "openapi.json")
+          @openapi_spec      = skeletal_openapi_spec
+        end
+
+        def generate!
+          openapi_spec["components"]["schemas"] = build_schemas
+          openapi_path.write("#{JSON.pretty_generate(openapi_spec)}\n")
+        end
+
+        private
+
+        def api_version
+          ManageIQ::Api::VERSION
+        end
+
+        def server_base_path
+          "/api(/:version)"
+        end
+
+        def build_schemas
+          schemas = {
+            "ID" => {
+              "type"        => "string",
+              "description" => "ID of the resource",
+              "pattern"     => "^\\d+$",
+              "readOnly"    => true,
+            }
+          }
+
+          ::Api::ApiConfig.collections.each do |_collection_name, collection|
+            next unless collection.klass
+
+            model       = collection.klass.constantize
+            schema_name = model.name.gsub("::", "_")
+
+            schemas[schema_name] = {
+              "type"                 => "object",
+              "properties"           => build_schema_properties(model),
+              "additionalProperties" => false
+            }
+          end
```

Comment:
After we build the schemas, can we sort by schema_name?  Right now they seem to come in a strange order.

Reply:
Or perhaps "ID" followed by our models sorted by name.

Reply:
:+1: yeah definitely, right now they are coming in in the order of the API yaml definition but alphabetically sorted is better